server/controllers/finance/account.js

```diff
@@ -104,6 +105,35 @@ function detail(req, res, next) {
   .done();
 }
 
+function getSold (req, res, next){
+  'use strict';
+
+  var accountId = req.params.id, optional = '';
+  var params = [accountId];
+
+  if(req.query.journal === '1') {
+    optional = ' UNION ALL SELECT pj.account_id, IFNULL(SUM(pj.debit_equiv), 0) AS debit, IFNULL(SUM(pj.credit_equiv), 0) AS credit, IFNULL((pj.debit_equiv - pj.credit_equiv), 0) AS balance FROM posting_journal AS pj WHERE pj.account_id = ? GROUP BY pj.account_id';
+    params.push(accountId);
+  }
+
+  var accountSoldQuery = 'SELECT t.account_id, IFNULL(SUM(t.debit), 0) AS debit, IFNULL(SUM(t.credit), 0) AS credit, IFNULL(t.balance, 0) AS balance FROM' + 
+    ' (SELECT gl.account_id, IFNULL(SUM(gl.debit_equiv), 0) AS debit, IFNULL(SUM(gl.credit_equiv), 0) AS credit, IFNULL((gl.debit_equiv - gl.credit_equiv), 0) AS balance FROM' + 
+    ' general_ledger AS gl WHERE gl.account_id = ? GROUP BY gl.account_id' + optional + ' ) AS t GROUP BY t.account_id';
+
+  lookupAccount(accountId, req.codes)
+    .then(function (account) {
+      return db.exec(accountSoldQuery, params);
+    })
+    .then(function (rows){
+      if(rows.length === 0){
```

Comment:
if there is no record for the provided account id in the ```posting_journal``` or the ```general_ledger```, I am sending back 404.
May be I should send this : ``` {account_id : x, debit : 0, credit : 0, balance : 0}```, what is your idea?